src/main/java/org/mockito/internal/configuration/plugins/DefaultMockitoPlugins.java

```diff
@@ -63,6 +65,8 @@ public class DefaultMockitoPlugins implements MockitoPlugins {
         DEFAULT_PLUGINS.put(
                 DoNotMockEnforcer.class.getName(),
                 "org.mockito.internal.configuration.DefaultDoNotMockEnforcer");
+        DEFAULT_PLUGINS.put(
+                MockCandidateFilter.class.getName(), TypeBasedCandidateFilter.class.getName());
```

Comment:
While type-safety is nice, we don't want to refer to the concrete implementations here so to avoid loading the classes if they are overridden. Therefore, let's use the string names here instead like the other plugin definitions.